index.js

```diff
@@ -9,31 +12,113 @@ function ask(questionText) {
+    signMsg: `The sign says "Welcome to Burlington Code Academy! Come on up to
+    the third floor. If the door is locked, use the code 12345."`, 
+    exitKey: `12345` },
+  'foyer':     {
+    canMoveTo: [ `stairway` ] ,
+    roomMessage: `You are in foyer`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit1` },
+  'stairway':  {
+    canMoveTo: [ `hallway` ],   
+    roomMessage: `You are in stairway`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit2` },
+  'hallway':   {
+    canMoveTo: [ 'classroom' ], 
+    roomMessage: `You are in hallway`,
+    signMsg: `This sign says nothing.`,                           
+    exitKey: `exit3` },
+  'classroom': {
+    canMoveTo: [ 'hallway' ],   
+    roomMessage: `You are in classroom!`,
+    signMsg: `This sign says "classroom".`,                       
+    exitKey: `exit4` }
 };
+let currentRoom = "outside";
 
-let currentState = "green";
+var player = {
+  currentState: currentRoom, 
+  inventory: []
+}
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
+function enterRoom(newRoom) {
+  let validTransitions = rooms[currentRoom].canMoveTo;
+  if (validTransitions.includes(newRoom)) {
+    currentRoom = newRoom;
   } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
+    throw 'Invalid room transition attempted - from ' + currentRoom + ' to ' + newRoom;
   }
-}
 
 start();
 
 async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+
+  const userPrompt = `\n>_`;
+  let userAnswer = await ask(rooms[currentRoom].roomMessage + userPrompt);
+  let userAnswerLC = userAnswer.toLowerCase(); 
+  while (userAnswerLC != 'exit') { 
+    if ((userAnswerLC == 'gargle') && (room[currentRoom] == 'outside')) {
+      console.log(`Sorry, I don't know how to ` + userAnswerLC + `.` + userPrompt);
+    } else if (userAnswerLC == 'read sign') {
+      console.log(signMsg + userPrompt);    
+    } else if (userAnswerLC == this.exitKey) {
+        enterRoom();
+      // return;  
+    } else if (userAnswerLC == 'exit') { 
+      console.log("Thanks for playing.");
+      process.exit();
+    } 
+    userAnswer = await ask(this.roomMessage + userPrompt); // keeps on asking
+    userAnswerLC = userAnswer.toLowerCase(); // keeps lowercasing the answer 
+  }
+}  // end of main pgm
+
+// functions should be last?
+
+  console.log(rooms[currentRoom].welcomeMessage);
+  //const userPrompt = `\n>_`;
+ // let userAnswerLC = userAnswer.toLowerCase(); 
+  let answer = await ask('_>');
```

Comment:
If you want the code below here to be separate from the start function (a totally valid decision) it needs to be wrapped in an async function, which will then have to be called at some point in the start function.